mbuild/tests/test_compound.py

```diff
@@ -68,6 +68,17 @@ def test_save_forcefield_with_file(self, methane):
                          forcefield_files=get_fn('methane_oplssaa.xml'),
                          overwrite=True)
 
+    @pytest.mark.skipif(not has_foyer, reason="Foyer is not installed")
+    def test_save_forcefield_with_file_foyerkwargs(self, methane):
+        exts = ['.hoomdxml']
+        foyerkwargs = {'assert_improper_params': True}
+        with pytest.raises(Exception):
```

Comment:
If you don't pass `foyerkwargs` and it should fail, let's actually do that `save` call to make sure passing it makes a difference.